xilem/src/view/zstack.rs

```diff
@@ -0,0 +1,219 @@
+// Copyright 2024 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    core::{
+        AppendVec, DynMessage, ElementSplice, Mut, SuperElement, View, ViewElement, ViewMarker,
+        ViewSequence,
+    },
+    Pod, ViewCtx,
+};
+use masonry::{
+    widget::{self, Alignment, WidgetMut},
+    Widget,
+};
+
+pub fn zstack<State, Action, Seq: ZStackSequence<State, Action>>(sequence: Seq) -> ZStack<Seq> {
+    ZStack {
+        sequence,
+        alignment: Alignment::default(),
+    }
+}
+
+pub struct ZStack<Seq> {
+    sequence: Seq,
+    alignment: Alignment,
+}
+
+impl<Seq> ZStack<Seq> {
+    #[must_use]
+    pub fn alignment(mut self, alignment: impl Into<Alignment>) -> Self {
+        self.alignment = alignment.into();
+        self
+    }
+}
+
+impl<Seq> ViewMarker for ZStack<Seq> {}
+impl<State, Action, Seq> View<State, Action, ViewCtx> for ZStack<Seq>
+where
+    State: 'static,
+    Action: 'static,
+    Seq: ZStackSequence<State, Action>,
+{
+    type Element = Pod<widget::ZStack>;
+
+    type ViewState = Seq::SeqState;
+
+    fn build(&self, ctx: &mut ViewCtx) -> (Self::Element, Self::ViewState) {
+        let mut elements = AppendVec::default();
+        let mut widget = widget::ZStack::new().with_alignment(self.alignment);
+        let seq_state = self.sequence.seq_build(ctx, &mut elements);
+        for child in elements.into_inner() {
+            widget = widget.with_child_pod(child.0.inner);
+        }
+        (ctx.new_pod(widget), seq_state)
+    }
+
+    fn rebuild(
+        &self,
+        prev: &Self,
+        view_state: &mut Self::ViewState,
+        ctx: &mut ViewCtx,
+        mut element: Mut<Self::Element>,
+    ) {
+        if self.alignment != prev.alignment {
+            widget::ZStack::set_alignment(&mut element, self.alignment);
+        }
+
+        let mut splice = StackSplice::new(element);
+        self.sequence
+            .seq_rebuild(&prev.sequence, view_state, ctx, &mut splice);
+        debug_assert!(splice.scratch.is_empty());
+    }
+
+    fn teardown(
+        &self,
+        view_state: &mut Self::ViewState,
+        ctx: &mut ViewCtx,
+        element: Mut<Self::Element>,
+    ) {
+        let mut splice = StackSplice::new(element);
+        self.sequence.seq_teardown(view_state, ctx, &mut splice);
+        debug_assert!(splice.scratch.into_inner().is_empty());
+    }
+
+    fn message(
+        &self,
+        view_state: &mut Self::ViewState,
+        id_path: &[crate::core::ViewId],
+        message: DynMessage,
+        app_state: &mut State,
+    ) -> crate::core::MessageResult<Action, DynMessage> {
+        self.sequence
+            .seq_message(view_state, id_path, message, app_state)
+    }
+}
+
+// MARK: ZStackElement
```

Comment:
This seems to be a different format to the other marker comments?

Reply:
Fixed